tests/testthat/test_wwinference.R

```diff
@@ -60,11 +60,15 @@ test_that("Function to get mcmc options produces the expected outputs", {
   mcmc_options <- get_mcmc_options()
   expected_names <- c(
     "iter_warmup", "iter_sampling",
-    "n_chains", "seed", "adapt_delta", "max_treedepth",
-    "compute_likelihood"
+    "chains", "seed", "adapt_delta", "max_treedepth",
+    "show_messages", "refresh", "save_latent_dynamics", "output_dir",
+    "output_basename", "sig_figs", "parallel_chains", "chain_ids",
+    "threads_per_chain", "opencl_ids", "save_warmup", "thin",
+    "adapt_engaged", "step_size", "metric", "metric_file", "inv_metric",
+    "init_buffer", "term_buffer", "window", "fixed_param", "show_exceptions",
+    "diagnostics", "save_metric", "save_cmdstan_config"
   )
-  # Checkmade doesn't work here for a list, says it must be a character vector
-  expect_true(all(names(mcmc_options) %in% expected_names))
+  checkmate::check_names(names(mcmc_options), must.include = expected_names)
```

Comment:
```suggestion
  checkmate::expect_names(names(mcmc_options), must.include = expected_names)
```

Reply:
note this is a stricter check than what you had before. Is it the one you want? Do you intend to have `get_mcmc_options()` define all those fields, or will it defer some to the `cmdstanr` defaults?

Reply:
Incidentally, for validation if you want to allow the user to set some but not all, but error if there are any unexpected arguments, you can use the `subset.of` argument to [`check/assert/expect_names()`](https://mllg.github.io/checkmate/reference/checkNames.html)

Reply:
True, I think now that we are using `do.call()` the list doesn't explicitly pass the args so it just needs to be a subset of all possible cmdstanr args, though as written it will pass all of them.

I could add another test that ensure that the `fit_opts` specified by the user are a `subset_of` the cmdstanr sample args?

Reply:
> True, I think now that we are using do.call() the list doesn't explicitly pass the args so it just needs to be a subset of all possible cmdstanr args, though as written it will pass all of them.

Correct, that's how it allows us more easily to defer some to the `cmdstanr` defaults, which was why I was suggesting above that `get_mcmc_options()` should maybe _not_ define all the fields, since there may be some where we'd prefer to defer to `cmdstanr` defaults.

> I could add another test that ensure that the fit_opts specified by the user are a subset_of the cmdstanr sample args?

Yes, that was what I was suggesting.

Reply:
I wrote in the cmdstanr defaults (bc initially I thought that was needed to make it configurable). Happy to remove all the ones we added that are literally just the defaults repeated.

Reply:
Yes, I would recommend that